store/RedisStore.js

```diff
@@ -1,50 +1,88 @@
 'use strict'
 
+const luaBasic = `
+  -- Key to operate on
+  local key = KEYS[1]
+  -- Time window for the TTL
+  local timeWindow = tonumber(ARGV[1])
+
+  -- Increment the key's value
+  local value = redis.call('INCR', key)
+
+  -- Check the current TTL of the key
+  local ttl = redis.call('PTTL', key)
+
+  -- If the key is new then set its TTL
+  if ttl == -1 then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {1, timeWindow}
+  end
+
+  return {value, ttl}
+`
+
+const luaContinueExceeding = `
+  -- Key to operate on
+  local key = KEYS[1]
+  -- Time window for the TTL
+  local timeWindow = tonumber(ARGV[1])
+  -- Max allowed value
+  local max = tonumber(ARGV[2])
+
+  -- Increment the key's value
+  local value = redis.call('INCR', key)
+
+  -- Check the current TTL of the key
+  local ttl = redis.call('PTTL', key)
+
+  -- If the key is new then set its TTL
+  if ttl == -1 then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {1, timeWindow}
+  -- If the key's incremented value has exceeded the max value, then reset its TTL
+  elseif value > max then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {value, timeWindow}
+  end
+
+  return {value, ttl}
+`
+
 function RedisStore (redis, timeWindow, continueExceeding, key) {
   this.redis = redis
   this.timeWindow = timeWindow
   this.continueExceeding = continueExceeding
   this.key = key
+
+  if (this.continueExceeding && !this.redis.rateLimitCE) {
+    this.redis.defineCommand('rateLimitCE', {
+      numberOfKeys: 1,
+      lua: luaContinueExceeding
+    })
+  } else if (!this.continueExceeding && !this.redis.rateLimit) {
+    this.redis.defineCommand('rateLimit', {
+      numberOfKeys: 1,
+      lua: luaBasic
+    })
+  }
```

Comment:
```suggestion
  this.redis = redis
  this.timeWindow = timeWindow
  // maybe undfined, but would be fine
  const oldContinueExceeding = this.continueExceeding
  this.continueExceeding = continueExceeding
  this.key = key
  
  if (!this.redis.rateLimit || oldContinueExceeding !== this.continueExceeding) {
    // we need to define rate limit command if
    // 1. not exist before
    // 2. continue exceeding value is changed
    this.redis.defineCommand('rateLimit', {
      numberOfKeys: 1,
      lua: this.continueExceeding ? luaContinueExceeding : luaBasic
    })
  }
```

Reply:
I don't know if redis allowed to redefine the command?

Reply:
Yeah that's what came to my mind at first, but here's why that won't work: redis is an object, and as you know when we redefine the `rateLimit` property, parent routes' `this.redis.rateLimit` will be updated as well, because they share the same object reference

This will essentially break any global config that sets `continueExceeding: false` if the last registered route has a config that enables continueExceeding or vice versa